### PR TITLE
MSPSDS-516: Report initial case correspondence

### PR DESCRIPTION
**Pull request by kicferk1**

### CODE REVIEW

**Comment by kicferk1** [comment]
I provided a rudimentary file view in a place where correspondence is displayed, but to truly reuse the functionality from other bits of code we probably need activities.

File: web/app/views/investigations/correspondence/general_info.html.slim [hunk 1]
```diff
@@ -33,44 +34,44 @@ main#main-content.govuk-main-wrapper role="main"
                 = form.label :correspondent_type, "Retailer", class: "govuk-label govuk-radios__label"
               .govuk-radios__item
                 = form.radio_button :correspondent_type, "Trading Standards / Other MSA", class: "govuk-radios__input"
-                = form.label :correspondent_type, "Trading Standards / Other MSA",\
```
Comment (broder) [inline web/app/views/investigations/correspondence/general_info.html.slim]: Are these `\` not required?

File: web/app/views/investigations/tabs/_details.html.slim [hunk 2]
```diff
@@ -40,16 +40,25 @@ p = link_to "Edit details", edit_investigation_path(@investigation), class: "no-
...
             tr.govuk-table__row
               th.govuk-table__header scope="row" Overview
               td.govuk-table__cell = correspondence.overview
             tr.govuk-table__row
               th.govuk-table__header scope="row" Details
               td.govuk-table__cell = correspondence.details
+            tr.govuk-table__row
+              th.govuk-table__header scope="row" File
+              td.govuk-table__cell
+                - correspondence.documents&.each { |doc|
```
Comment (broder) [inline web/app/views/investigations/tabs/_details.html.slim]: I'm not sure that you need the curly braces if you indent the lines within the block

File: web/app/models/audit_activity/correspondence/base.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class AuditActivity::Correspondence::Base < AuditActivity::Base
+  belongs_to :correspondence
```
Comment (broder) [inline web/app/models/audit_activity/correspondence/base.rb]: Also, do we actually need to persist this relationship? Is there a link from the activity to the correspondence tab / page?

File: web/app/views/investigations/correspondence/general_info.html.slim (hunk 1 above)
Comment (broder) [reply web/app/views/investigations/correspondence/general_info.html.slim]: Neat!

File: web/app/views/investigations/tabs/_details.html.slim (hunk 2 above)
Comment (kicferk1) [reply web/app/views/investigations/tabs/_details.html.slim]: Sadly you get a syntax error if you don't provide a proc to .each :( So either do end, or { } I think

File: web/app/models/audit_activity/correspondence/base.rb (hunk 3 above)
Comment (kicferk1) [reply web/app/models/audit_activity/correspondence/base.rb]: That's probably a design question, there are links on correspondence activities saying 'view more'. If it opens a new page we probably need it, if it just expands the text field then we probably don't. 

Tim says we will probably need it in the future though.

File: web/app/models/audit_activity/correspondence/add.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class AuditActivity::Correspondence::Add < AuditActivity::Correspondence::Base
+  def self.from(correspondence, investigation)
```
Comment (jasiekmiko) [inline web/app/models/audit_activity/correspondence/add.rb]: I don't think we gain much from repeating this in all of the activities, it's not actually doing anything...
Worth cutting this?

File: web/test/fixtures/sources.yml [hunk 5]
```diff
@@ -39,3 +39,7 @@ address_one:
 address_one:
   type:
   sourceable: two (Address)
+
+corresopndence_one:
```
Comment (jasiekmiko) [inline web/test/fixtures/sources.yml]: Can we fill it in with more data to make it more useful down the line? e.g. name, created at, user_id?

File: web/test/system/add_correspondence_from_flow_test.rb [hunk 6]
```diff
@@ -11,19 +11,45 @@ class AddCorrespondenceFromFlowTest < ApplicationSystemTestCase
...
     visit root_path
     click_on "Report an unsafe product"
     select_type_and_continue
     fill_name_and_continue
-    click_on "Add reporter correspondance"
+    click_on "Add reporter correspondence"
     assert_equal('Ben', find_field('correspondence[correspondent_name]').value)
   end
 
+  test "first step should be populated with email from the flow" do
```
Comment (jasiekmiko) [inline web/test/system/add_correspondence_from_flow_test.rb]: "... populated with reporter's email" perhaps?

File: web/app/views/investigations/tabs/_details.html.slim [hunk 7]
```diff
@@ -40,16 +40,25 @@ p = link_to "Edit details", edit_investigation_path(@investigation), class: "no-
...
-              - if correspondence.contact_method == 'Phone call'
-                td.govuk-table__cell = "#{correspondence.contact_method}: #{correspondence.phone_number}"
+              td.govuk-table__cell
+                - if correspondence.contact_method == 'email'
+                  = "Email: #{correspondence.email_address}"
+                - if correspondence.contact_method == 'phone'
+                  = "Phone call: #{correspondence.phone_number}"
             tr.govuk-table__row
-              th.govuk-table__header scope="row" Correspondent name
+              th.govuk-table__header scope="row" Correspondent date
```
Comment (jasiekmiko) [inline web/app/views/investigations/tabs/_details.html.slim]: "Correspondence"

**Pull request merged**